apps/system/js/accessibility.js

```diff
@@ -284,11 +290,33 @@
     },
 
     /**
+     * Start a timeout that waits to display hints
+     * @memberof Accessibility.prototype
+     */
+    setHintsTimeout: function ar_setHintsTimeout(aDetails, aHints) {
+        clearTimeout(this.hintsTimer);
```

Comment:
Had to add this here to avoid overlapping Timeouts... I think the problem stems from the fact that when a speech action is canceled, its "end" events still fire anyway, so this.hintsTimer gets overwritten and you have two simultaneous loops.  Not sure if this is what you were looking for here.

Additionally, were you just wanting this loop to keep suggesting things forever?  Or did you want it to stop after a certain number of suggestions?